flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/BatchingStateChangeUploader.java

```diff
@@ -121,18 +127,17 @@ public void upload(UploadTask uploadTask) {
         }
         LOG.debug("persist {} changeSets", uploadTask.changeSets.size());
         try {
-            checkState(
-                    inFlightBytesCounter.sum() <= maxBytesInFlight,
-                    "In flight data size threshold exceeded %s > %s",
-                    inFlightBytesCounter.sum(),
-                    maxBytesInFlight);
+            long size = uploadTask.getSize();
+            uploadThrottle.seizeCapacity(size);
             synchronized (scheduled) {
-                long size = uploadTask.getSize();
-                inFlightBytesCounter.add(size);
                 scheduledBytesCounter += size;
-                scheduled.add(wrapWithSizeUpdate(uploadTask, size, inFlightBytesCounter));
+                scheduled.add(wrapWithSizeUpdate(uploadTask, size, uploadThrottle));
                 scheduleUploadIfNeeded();
             }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
did you mean `interrupted()` here?

Reply:
No, I actually mean `interrupt()` to restore interrupt status (since `InterruptedException` is caught and wrappen, and caller might not check the cause but might check the status).